From f6b02e4204cecad351304099334be6ec1bf7a934 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Wed, 14 Nov 2018 15:02:21 +0700
Subject: [PATCH] Problem: authorship derivation can't write to a config file

This happens when the config directory doesn't exist.

Solution: during sit's startup, ensure config dir exists
---
 sit/src/main.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sit/src/main.rs b/sit/src/main.rs
index 07b112bb..6b5047ea 100644
--- a/sit/src/main.rs
+++ b/sit/src/main.rs
@@ -430,6 +430,7 @@ fn main_with_result(allow_external_subcommands: bool) -> i32 {
     let default_config = project_dirs.config_dir().join("config.json");
 
     let config_path = matches.value_of("config").unwrap_or(default_config.to_str().unwrap());
+    std::fs::create_dir_all(project_dirs.config_dir()).expect("can't ensure config directory's presence");
 
     let mut settings = config::Config::default();
     settings
-- 
2.18.1

